Bellerophon/Bellerophon/Sources/BPManagerDelegate.swift

```diff
@@ -7,42 +7,56 @@
 //
 
 /// The Bellerophon manager delegate.
-@objc public protocol BellerophonManagerDelegate {
+public protocol BellerophonManagerDelegate: class {
 
+    // MARK: - Required
     /**
      Provide Bellerophon's current status according to the status object received
 
      - parameter manager:    The Bellerophon manager.
      - parameter completion: The completion block.
      */
-    func bellerophonStatus(_ manager: BellerophonManager,
-                           completion: @escaping (_ status: BellerophonObservable?,
-        _ error: NSError?) -> ())
+    func bellerophonStatus(_ manager: BellerophonManager, completion: @escaping (_ status: BellerophonObservable?, _ error: Error?) -> ())
+
+
+    // MARK: - Optional
+    /// The app is notified when a error was received.
+    ///
+    /// - Parameter error: The error received.
+    func receivedError(error: Error)
 
     /**
      The app is notified that a force update should occur. An alert should be displayed to redirect to the App Store.
      */
     func shouldForceUpdate()
 
-    /// The app is notified when a error was received.
-    ///
-    /// - Parameter error: The error received.
-    func receivedError(error: NSError)
+    /**
+     The app is notified that a kill switch should occur.
+     */
+    func shouldKillSwitch()
```

Comment:
What is the purpose of this function? Aren't we presenting the kill switch window anyway? I'm assuming this is to cover custom logic when the kill switch is activated. Although, do we have a duplication of behavior with `bellerophonWillEngage`?

Reply:
This method is similar to `shouldForceUpdate`. Developers are no longer required to provide a kill switch view. This gives the option of responding to the kill switch event without having Bellerophon display a kill switch view. `bellerophonWillEngage` will only be called if Bellerophon presents a window to the user. If a developer decides to provide a kill switch view and handle the call back for `shouldKillSwitch`, then it would be duplicated.